### PR TITLE
Attempt to decorate k8 resource manual deletion with message

### PR DESCRIPTION
**Pull request by Ln11211**
# TL;DR
This is an attempt to provide a reason for the pod failure when the resources are manually deleted.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This is just an attempt to provide a reason for pod failure when the k8 resources are deleted manually, I am unsure about this fix. I used the **k8serrors.ReasonForError()** to get the reason for pod failure and then attached it to the object that failed. Check [line 317](https://github.com/Ln11211/flyteplugins/blob/ceee0fa0a881ec2a394b03fe2fe5ebad8b36022c/go/tasks/plugins/array/k8s/subtask.go#L317) and 318.

## Tracking Issue
Issue [#2843](https://github.com/flyteorg/flyte/issues/2843)

### CODE REVIEW

**Comment by Ln11211** [comment]
Hello @hamersaw , I attempted at solving issue [#2843](https://github.com/flyteorg/flyte/issues/2843) . Do you think this will work?

**Comment by Ln11211** [comment]
hello @hamersaw, what do you think about this approach?, will this be sufficient?, can we get the reason for the pod deletions this way?

**Comment by EngHabu** [comment]
Hey @Ln11211, the idea is to use pod.reason and pod.message fields (check out https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/ for where to find the fields) and copy these into the error message..

**Comment by Ln11211** [comment]
> Hey @Ln11211, the idea is to use pod.reason and pod.message fields (check out https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/ for where to find the fields) and copy these into the error message..

Hello there, Yes, I went through this page previously but didn't understand how to use those fields so, I read the v1 Core docs and found about the `Message` and `Reason` fields in the [PodStatus documentation](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#PodStatus-v1-core) and used them now, I guess this should work. Check [line 317](https://github.com/flyteorg/flyteplugins/blob/1446f76b873e887a33b265799a95df9b7074b470/go/tasks/plugins/array/k8s/subtask.go#L317).

**Comment by Ln11211** [comment]
hello @hamersaw , can I get a review please :)

**Comment by Ln11211** [comment]
hi @samhita-alla , would it be possible to review this PR?, This PR aims to provide more context for the manually deleted objects using the Pod `Message` and `Reason` fields :)

**Comment by hamersaw** [comment]
@Ln11211 were you able to test this and confirm the message and reason were correctly populated? Also, did you look into this fix in the FlytePropeller code? it seems this is the easy case (k8s-array subtasks), but the [general case](https://github.com/flyteorg/flytepropeller/blob/aa3e9a6dae446a3a4cbb7e92faa37d2ba75883a0/pkg/controller/nodes/task/k8s/plugin_manager.go#L306) is more difficult.
